### DISCUSSION TITLE
Changing the clip speed results in the clip no longer being selected

### DISCUSSION THREAD

**Discussion by DavidBailes** [audacity/audacity#6167] [Development]
@LWinterberg @saintmatthieu .
In 3.5 alpha, when you change the clip speed using the pitch and speed dialog, the time selection is not updated so that its length is the new length of the clip. This means that the clip is no longer selected. This is a change from the behaviour of the clip speed dialog in 3.4. What was the motivation for this change?

If you select an object, and open a dialog and change one of its properties, the standard behaviour is for the object to remain selected. The user wouldn't expect the object to be no longer selected.

For example, if the user selects a clip, reduces its speed using the pitch and speed dialog, presses Ctrl + C to copy, positions the cursor where they want to paste, and then presses Ctrl + V, I think that they would rightly expect to have pasted a copy of all the original clip (with its reduced speed). But this doesn't happen.

In contrast, if you destructively change the speed of a selected clip using the tempo effect, the clip remains selected.

**Comment by saintmatthieu** [comment]
Hi David,
Thanks for reporting this, and I would agree with you. Only now that a clip doesn't need to be completely selected for the P&S dialog to be invokable, there would be a need for a new rule, which probably should be: the selection is adjusted if the clip is exactly selected. Does that sound ok?

**Comment by DavidBailes** [comment]
> Hi David, Thanks for reporting this, and I would agree with you. Only now that a clip doesn't need to be completely selected for the P&S dialog to be invokable,

Looks like selection start has to be within the clip for the dialog to open.

> there would be a need for a new rule, which probably should be: the selection is adjusted if the clip is exactly selected. Does that sound ok?

I'm not sure. In the case that selection start and selection end are within the clip, I think the user might expect these to be adjusted so that they are at the same corresponding times in the audio. For example, starting from:
![image](https://github.com/audacity/audacity/assets/11887021/65402f62-ddaf-4a37-a5e8-653c1df57788)

After changing the speed to 50%:
![image](https://github.com/audacity/audacity/assets/11887021/7308c23c-8a45-4176-8370-49c461c5fa02)

Wouldn't the user expect the time selection to have changed to 8 - 12s ?

Maybe that's a matter of opinion. I think the important case is that the selection is adjusted if the clip is exactly selected.

**Comment by saintmatthieu** [comment]
Me too. https://github.com/audacity/audacity/issues/6169
